qiskit/transpiler/passes/optimization/split_2q_unitaries.py

```diff
@@ -24,17 +25,39 @@ class Split2QUnitaries(TransformationPass):
     matrix is actually a product of 2 single qubit gates. In these cases the 2q gate can be
     simplified into two single qubit gates and this pass will perform this optimization and will
     replace the two qubit gate with two single qubit :class:`.UnitaryGate`.
+
+    If some of the gates can be viewed as a swap joined by the product of 2 single qubit gates,
+    the pass will recreate the DAG, permuting the swapped qubits similar
+    to how it's done in :class:`ElidePermutations`.
     """
 
-    def __init__(self, fidelity: float = 1.0 - 1e-16):
+    def __init__(self, fidelity: float = 1.0 - 1e-16, split_swap: bool = False):
         """
         Args:
             fidelity: Allowed tolerance for splitting two-qubit unitaries and gate decompositions.
+            split_swap: Whether to attempt to split swap gates, resulting in a permutation of the qubits.
         """
         super().__init__()
         self.requested_fidelity = fidelity
+        self.split_swap = split_swap
 
     def run(self, dag: DAGCircuit) -> DAGCircuit:
         """Run the Split2QUnitaries pass on `dag`."""
-        split_2q_unitaries(dag, self.requested_fidelity)
-        return dag
+        result = split_2q_unitaries(dag, self.requested_fidelity, self.split_swap)
+        if result is None:
+            return dag
+
+        (new_dag, qubit_mapping) = result
+        input_qubit_mapping = {qubit: index for index, qubit in enumerate(dag.qubits)}
+        self.property_set["original_layout"] = Layout(input_qubit_mapping)
```

Comment:
Does this behave correctly if we run elide permutations first? I realize this is just a straight copy of what elide permutations is doing, but I'm wondering if we need to check that we're not doing multiple permutations before applying the layout or do some other storing if we run these together.

Reply:
It should work - in theory. in `split_2q_unitaries.py` we have
```
if current_layout := self.property_set["virtual_permutation_layout"]:
            self.property_set["virtual_permutation_layout"] = new_layout.compose(
                current_layout, dag.qubits
            )
```

And `compose` works, if I recall correctly. However, this is not covered by the current test suite, and that's one of the things I wanted to ask about - should the "joint" test be placed in `TestSplit2QUnitaries` or perhaps a different location? What flows should we check, other than adding `Split2QUnitaries` before and after `ElidePermutations`?